### ISSUE TITLE
create alert rule failed

### ISSUE THREAD

**Issue by jojotong** [kubegems/kubegems#340]
error msg:
```
failed to get channel by receiver: {0xc0035efa90 1m {aaa-id-23 [] [] [] [{0xc0035f7c19 0xc000cfd330 nil 0xc003c0cc00 0}] [] [] [] []} 2}: channel: 23 not found
```
It's because some channel id `23` is deleted, by the alert rule exist.
```yaml
  - name: aaa-id-23
    webhookConfigs:
    - httpConfig:
        tlsConfig:
          ca: {}
          cert: {}
          insecureSkipVerify: true
      sendResolved: false
      url: https://baodu.com
```

**Comment by jojotong** [comment]
Resolved by delete the alert rule which use channel `23`.
But should resolved totally after #329

**Comment by jojotong** [comment]
Fixed by #371
